add two test applications

bollhals opened the pull request:
## Proposed Changes

adds two test applications to execute when you want to profile (performance or memory) some changes.
I think it would be good to have them in git so everyone has access and we're talking about the same thing.

Apps added:
- Message batch publish / receive (originated from @stebet)
- Open / Close Channels

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes issue #NNNN)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [ ] Cosmetic change (whitespace, formatting, etc)

## Checklist

- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [x] All tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in related repositories

michaelklishin commented:
We have a benchmarks project now, is this really different from why it was added?

bollhals commented:
> We have a benchmarks project now, is this really different from why it was added?

If you want to speifically analyze e.g what allocation happend (the images shown in the past PRs) you need an small application to profile. 

The benchmark only yields overall results, but won't give you details about whether or not the thing you tried to eliminate is still allocated. 

Different apps because you may want to look at different things. I've added the batch publish one because that one we used since @stebet used it to track allocations. I've added the channel one because I wanted to see what exactly gets allocated when we crate a channel.
